opensrp-reveal/src/main/java/org/smartregister/reveal/presenter/ListTaskPresenter.java

```diff
@@ -328,9 +319,9 @@ public void onFormSaved(@NonNull String structureId, @NonNull TaskStatus taskSta
         listTaskView.setGeoJsonSource(featureCollection, null);
 
         if (IRS.equals(interventionType)) {
-            listTaskInteractor.fetchSprayDetails(structureId, false);
-        } else if (MOSQUITO_COLLECTION.equals(interventionType)) {
-            listTaskInteractor.fetchMosquitoCollectionDetails(structureId, false);
+            listTaskInteractor.fetchInterventionDetails(IRS, structureId, false);
```

Comment:
Maybe you can remove this condition and merge to  
`listTaskInteractor.fetchInterventionDetails(interventionType, structureId, false);`

Reply:
Done.